source/3_Netplugin.md

```diff
@@ -0,0 +1,180 @@
+# Netplugin
+
+## About
+Contiv netplugin is a generic networking plugin that is designed to provide multi host policy based networking for containerized applications.
+
+Netplugin is designed to be used with docker containers and with cluster schedulers like Swarm, Kubernetes and Mesos
+
+## Getting started
+This section will walk you through how to try netplugin in a vagrant environment
+### Prerequisits
+- VirtualBox 5.0.2 or greater
+- Vagrant 1.7.4
+- Go 1.5.0
+
+### Quick start guide
+Clone the repo and bring up the VMs
+```
+$ git clone https://github.com/contiv/netplugin
+$ cd netplugin; make demo
+$ vagrant ssh netplugin-node1
+```
+This brings up a cluster of two VMs running docker and netplugin
+
+#### Step 1: Create a network
+```
+netplugin-node1$ netctl net create contiv-net --subnet=20.1.1.0/24 --gateway=20.1.1.254 --pktTag=200
+```
+
+#### Step 2: Run containers on two hosts
+On node1
+```
+netplugin-node1$ docker run -itd --name=web --net=contiv-net ubuntu /bin/bash
+```
+
+On node2
+```
+netplugin-node2$ docker run -itd --name=db --net=contiv-net ubuntu /bin/bash
+```
+
+On node1, login to the container and ping the container on node2
+```
+netplugin-node1$ docker exec -it web /bin/bash
+< inside the container >
+root@f90e7fd409c4:/# ping db
+PING db (20.1.1.3) 56(84) bytes of data.
+64 bytes from db (20.1.1.3): icmp_seq=1 ttl=64 time=0.658 ms
+64 bytes from db (20.1.1.3): icmp_seq=2 ttl=64 time=0.103 ms
+```
+
+
+## Architecture
+#### Glossary
+- *Netmaster*: Netmaster is the master daemon that is responsible for storing the Intent and distributing state to all nodes in the cluster.
+- *Netplugin*: Netplugin is a long running daemon on every node in the cluster. As a docker network plugin, it is responsible for setting up network and policy for each container.
+-  *Netctl*: netctl is a command line tool to moidfy the intent.
+-  *Intent*: <FIXME: define intent>
```

Comment:
That's because they're invalid HTML, which is legal in markdown. They're 
just not being displayed by your web browser.

On 15 Nov 2015, at 4:13, Sukhesh Halemane wrote:

>> +```
>> +netplugin-node1$ docker exec -it web /bin/bash
>> +< inside the container >
>> +root@f90e7fd409c4:/# ping db
>> +PING db (20.1.1.3) 56(84) bytes of data.
>> +64 bytes from db (20.1.1.3): icmp_seq=1 ttl=64 time=0.658 ms
>> +64 bytes from db (20.1.1.3): icmp_seq=2 ttl=64 time=0.103 ms
>> +```
>> +
>> +
>> +## Architecture
>> +#### Glossary
>> +- *Netmaster*: Netmaster is the master daemon that is responsible 
>> for storing the Intent and distributing state to all nodes in the 
>> cluster.
>> +- *Netplugin*: Netplugin is a long running daemon on every node in 
>> the cluster. As a docker network plugin, it is responsible for 
>> setting up network and policy for each container.
>> +-  *Netctl*: netctl is a command line tool to moidfy the intent.
>> +-  *Intent*: <FIXME: define intent>
>
> sure. i'll remove it.
> btw, <FIXME> tags are not visible in markdown. its mostly for our 
> reference
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/contiv/docs/pull/1/files#r44866961